diff-containers/test/Test/Data/Map/Diff/Strict.hs

```diff
@@ -174,3 +273,78 @@ instance Arbitrary v => Arbitrary (DiffEntry v) where
     Delete x           -> Delete <$> shrink x
     UnsafeAntiInsert x -> UnsafeAntiInsert <$> shrink x
     UnsafeAntiDelete x -> UnsafeAntiDelete <$> shrink x
+
+{------------------------------------------------------------------------------
+  Modifiers: positivity
+------------------------------------------------------------------------------}
+
+-- | A @'Diff'@ for which @'isPositive'@ holds.
+newtype PositiveDiff k v = PositiveDiff { getPositiveDiff :: Diff k v }
+  deriving stock (Show, Eq)
+  deriving newtype (Semigroup, Monoid, Group)
+
+instance (Ord k, Arbitrary k, Arbitrary v)
+      => Arbitrary (PositiveDiff k v) where
+  arbitrary               = PositiveDiff <$> arbitrary `suchThat` isPositive
+  shrink (PositiveDiff d) = PositiveDiff <$> shrink d
+
+-- | A @'DiffHistory'@ for which @'isPositiveDiffHistory'@ holds.
+newtype PositiveDiffHistory v = PositiveDiffHistory {
+    getPositiveDiffHistory :: DiffHistory v
+  }
+  deriving stock (Show, Eq)
+  deriving newtype (Semigroup, Monoid, Group)
+
+instance Arbitrary v => Arbitrary (PositiveDiffHistory v) where
+  arbitrary = PositiveDiffHistory <$> arbitrary `suchThat` isPositiveDiffHistory
+  shrink (PositiveDiffHistory h) = PositiveDiffHistory <$> shrink h
+
+{------------------------------------------------------------------------------
+  Modifiers: normality
```

Comment:
Perhaps we want to test the generators in this section.

Reply:
Will do